stm/stm.ml

```diff
@@ -2234,10 +2234,13 @@ let known_state ?(redefine_qed=false) ~cache id =
 	    if eff then update_global_env ()
           ), (if eff then `Yes else cache), true
       | `Cmd { cast = x; ceff = eff } -> (fun () ->
-            reach view.next;
-            stm_vernac_interp id x;
-	    if eff then update_global_env ()
-          ), (if eff then `Yes else cache), true
+          (match !Flags.async_proofs_mode with
```

Comment:
I don't have enough time now (I'm on VAC) but consider:
- `-quick` implies `BuildVio`
- even if quick is not given one can pass to coqc `-async-proofs on` (to process proofs in parallel, as CoqIDE does)
- there are already flags to turn on/off error resiliency

Maybe the right fix is to have such flags be off (be default) in batch mode (Stm has a a function to check that, IIRC `interactive`).

Anyway, KUDOS for your understanding of this issue!